examples/java-matter-controller/java/src/com/matter/controller/commands/common/MatterCommand.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package com.matter.controller.commands.common;
+
+import com.matter.controller.config.PersistentStorage;
+import com.matter.controller.config.PersistentStorageOpCertStore;
+import com.matter.controller.config.PersistentStorageOperationalKeystore;
+import java.util.Optional;
+
+public abstract class MatterCommand extends Command {
+
+  public static final short kMaxGroupsPerFabric = 5;
+  public static final short kMaxGroupKeysPerFabric = 8;
+
+  protected PersistentStorage mDefaultStorage = new PersistentStorage();
+  protected PersistentStorage mCommissionerStorage = new PersistentStorage();
+  protected PersistentStorageOperationalKeystore mOperationalKeystore =
+      new PersistentStorageOperationalKeystore();
+  protected PersistentStorageOpCertStore mOpCertStore = new PersistentStorageOpCertStore();
+
+  protected Optional<CredentialsIssuer> mCredIssuerCmds;
+  protected StringBuffer mCommissionerName = new StringBuffer();
+  protected StringBuffer mPaaTrustStorePath = new StringBuffer();
+  protected StringBuffer mCDTrustStorePath = new StringBuffer();
+  protected Optional<Long> mCommissionerNodeId;
+  protected Optional<Short> mBleAdapterId;
+  protected Optional<Boolean> mUseMaxSizedCerts;
+  protected Optional<Boolean> mOnlyAllowTrustedCdKeys;
```

Comment:
Should these be assigned somewhere? Right now these will default to null (and mixing null and optional is error-prone). 

Perhaps these can be declared as nullable instead?

```
@Nullable private Long mCommissionerNodeId;
// etc
```